master/internal/rm/kubernetesrm/informer.go

```diff
@@ -43,38 +46,102 @@ func newInformer(
 
 	// Log when pods are first added to the informer (at start-up).
 	syslog := logrus.WithFields(logrus.Fields{
-		"component": "podInformer",
+		"component": fmt.Sprintf("%s-informer", name),
 		"namespace": namespace,
 	})
 	for i := range pods.Items {
-		syslog.Debugf("informer added pod: %s", pods.Items[i].Name)
-		cb(&pods.Items[i])
+		syslog.Debugf("informer added %s: %s", name, pods.Items[i].Name)
+		cb(watch.Event{Object: &pods.Items[i]})
 	}
 
 	return &informer{
 		cb:         cb,
+		name:       name,
 		syslog:     syslog,
 		resultChan: rw.ResultChan(),
 	}, nil
 }
 
-// startInformer returns the updated pod, if any.
-func (i *informer) run() {
-	i.syslog.Debug("pod informer is starting")
+func newEventInformer(
+	ctx context.Context,
+	eventInterface typedV1.EventInterface,
+	namespace string,
+	cb informerCallback,
+) (*informer, error) {
+	events, err := eventInterface.List(ctx, metaV1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	rw, err := watchtools.NewRetryWatcher(events.ResourceVersion, &cache.ListWatch{
+		WatchFunc: func(options metaV1.ListOptions) (watch.Interface, error) {
+			return eventInterface.Watch(ctx, metaV1.ListOptions{})
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// Log when pods are first added to the informer (at start-up).
+	syslog := logrus.WithFields(logrus.Fields{
+		"component": "event-informer",
+		"namespace": namespace,
+	})
+	for i := range events.Items {
+		syslog.Debugf("informer added event: %s", events.Items[i].Name)
+		cb(watch.Event{Object: &events.Items[i]})
+	}
+
+	return &informer{
+		cb:         cb,
+		name:       "event",
+		syslog:     syslog,
+		resultChan: rw.ResultChan(),
+	}, nil
+}
+
+func newNodeInformer(
+	ctx context.Context,
+	nodeInterface typedV1.NodeInterface,
+	cb informerCallback,
+) (*informer, error) {
+	nodes, err := nodeInterface.List(ctx, metaV1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	rw, err := watchtools.NewRetryWatcher(nodes.ResourceVersion, &cache.ListWatch{
+		WatchFunc: func(options metaV1.ListOptions) (watch.Interface, error) {
+			return nodeInterface.Watch(ctx, options)
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// Log when nodes are first added to the informer (at start-up).
+	syslog := logrus.WithField("component", "nodeInformer")
+	for i := range nodes.Items {
+		syslog.Debugf("informer added node: %s", nodes.Items[i].Name)
+		cb(watch.Event{Object: &nodes.Items[i], Type: watch.Added})
+	}
+
+	return &informer{
+		cb:         cb,
+		name:       "node",
+		syslog:     syslog,
+		resultChan: rw.ResultChan(),
+	}, nil
+}
+
+func (i *informer) run(ctx context.Context) {
```

Comment:
@stoksc in another PR, you suggested adding in a context to `run()` to facilitate cancellation in the event there are several unrelated goroutines in the stack -- what would that actually look like in this function? adding something to the panic message?